code/modules/mob/living/carbon/human/human.dm

```diff
@@ -295,13 +295,19 @@
 	var/obj/item/organ/external/affecting = get_organ(ran_zone(dam_zone))
 	apply_damage(5, BRUTE, affecting, run_armor_check(affecting, "melee"))
 
-/mob/living/carbon/human/bullet_act()
-	if(mind && mind.martial_art && mind.martial_art.deflection_chance) //Some martial arts users can deflect projectiles!
+/mob/living/carbon/human/bullet_act(obj/item/projectile/P, def_zone)
+	if(mind && mind.martial_art && mind.martial_art.deflection_chance  && in_throw_mode) //Some martial arts users can deflect projectiles!
 		if(!prob(mind.martial_art.deflection_chance))
 			return ..()
 		if(!src.lying && !(HULK in mutations)) //But only if they're not lying down, and hulks can't do it
 			visible_message("<span class='danger'>[src] deflects the projectile; [p_they()] can't be hit with ranged weapons!</span>", "<span class='userdanger'>You deflect the projectile!</span>")
-			return 0
+			playsound(src, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, 1)
+			if(mind.martial_art.reroute_deflection)
+				P.firer = src
+				P.setAngle(rand(0, 360))
+				return -1
```

Comment:
The proc seems to return a boolean, so I don't think returning `-1` is the best idea here.